pkg/virt-controller/services/multus_annotations.go

```diff
@@ -54,11 +56,11 @@ func (mnap multusNetworkAnnotationPool) toString() (string, error) {
 	return string(multusNetworksAnnotation), nil
 }
 
-func GenerateMultusCNIAnnotation(namespace string, interfaces []v1.Interface, networks []v1.Network) (string, error) {
-	return GenerateMultusCNIAnnotationFromNameScheme(namespace, interfaces, networks, namescheme.CreateHashedNetworkNameScheme(networks))
+func GenerateMultusCNIAnnotation(namespace string, interfaces []v1.Interface, networks []v1.Network, config *virtconfig.ClusterConfig) (string, error) {
+	return GenerateMultusCNIAnnotationFromNameScheme(namespace, interfaces, networks, namescheme.CreateHashedNetworkNameScheme(networks), config)
 }
 
-func GenerateMultusCNIAnnotationFromNameScheme(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string) (string, error) {
+func GenerateMultusCNIAnnotationFromNameScheme(namespace string, interfaces []v1.Interface, networks []v1.Network, networkNameScheme map[string]string, config *virtconfig.ClusterConfig) (string, error) {
```

Comment:
I think this method is used when an interface is [hotplugged](https://github.com/kubevirt/kubevirt/blob/3541fa27801cb607c62b951869755f2cb5ee3705/pkg/virt-controller/watch/vmi.go#L2295) as well.

As we don't support hotplugging a side car, I don't think that in this stage we should support hotplugging a CNI.

Reply:
Isn't the hotplug limited to the core bridge binding?

Reply:
> As we don't support hotplugging a side car, I don't think that in this stage we should support hotplugging a CNI.
Done
I had to do some refactoring around multus annotation generating code to enable more granularity.

> Isn't the hotplug limited to the core bridge binding?

It is, but since the pod may end up with additional interface

Reply:
In case of interface hotplug where the interface spesifes a binding plugin and the plugin has a NAD registered, virt controller VMI controller will the pod Multus annotation and it will have the binding NAD.

Reply:
I'm confused.
We are not doing this for other bindings, e.g. masquerade, macvtap, passt, etc. We are not filtering them, and I do not understand how this is different.

We already depend on something else to filter the requests.
The only reason we had the SR-IOV check is that it depends on the actual state. That is a different logic.

So, if you think this is really needed, I would expect this to be done to all other binding types.